pkg/pgmodel/migrations/sql/1_base_schema.up.sql

```diff
@@ -1,13 +1,66 @@
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_PROM; -- data tables + public functions
+GRANT USAGE ON SCHEMA SCHEMA_PROM TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_PROM TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_PROM GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_PROM TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_PROM TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_PROM GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_EXT; -- optimized versions of functions created by the extension
+GRANT USAGE ON SCHEMA SCHEMA_EXT TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_EXT TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_EXT GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_EXT TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_EXT TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_EXT GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_SERIES;
+GRANT USAGE ON SCHEMA SCHEMA_SERIES TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_SERIES TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_SERIES GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_SERIES TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_SERIES TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_SERIES GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_METRIC;
+GRANT USAGE ON SCHEMA SCHEMA_METRIC TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_METRIC TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_METRIC GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_METRIC TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_METRIC TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_METRIC GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_DATA;
+GRANT USAGE ON SCHEMA SCHEMA_DATA TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_DATA TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_DATA GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_DATA TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_DATA TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_DATA GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
 CREATE SCHEMA IF NOT EXISTS SCHEMA_INFO;
+GRANT USAGE ON SCHEMA SCHEMA_INFO TO prom_reader;
+GRANT SELECT ON ALL TABLES IN SCHEMA SCHEMA_INFO TO prom_reader;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_INFO GRANT SELECT ON TABLES TO prom_reader;
+GRANT CREATE ON SCHEMA SCHEMA_INFO TO prom_writer;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA SCHEMA_INFO TO prom_writer;
+ALTER DEFAULT PRIVILEGES IN SCHEMA SCHEMA_INFO GRANT ALL PRIVILEGES ON TABLES TO prom_writer;
+
+GRANT prom_reader TO prom_writer;
```

Comment:
Doesn't horribly matter, but I'd put the grant of prom reader to prom writer right after you create the roles, just makes everything else make more sense.